views/js/previewer/proxy/item.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test runner proxy for the QTI item previewer
+ *
+ * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'core/promise',
+    'core/promiseQueue',
+    'core/communicator',
+    'taoQtiTest/previewer/config/item',
+    'util/httpErrorParser'
+], function($, _, __, Promise, promiseQueue, communicatorFactory, configFactory, httpErrorParser) {
```

Comment:
Shouldn't we have a very simple proxy that handles the only 2 requests we need ?

Reply:
Since the previewer is based on the test runner architecture and life cycle, I am not sure we can simplify the proxy so much